### PR TITLE
Add some minimal tests for cudnn softmax.

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

File: theano/sandbox/cuda/tests/test_dnn.py [hunk 1]
```diff
@@ -489,6 +489,25 @@ def test_log_softmax():
...
+    softmax_op = dnn.GpuDnnSoftmax('bc01', 'accurate', 'channel')
+
+    x_val = numpy.random.normal(0, 1, (3, 4, 1, 1)).astype('float32')
+
+    utt.verify_grad(softmax_op, [x_val])
+
+
+def test_dnn_softmax_grad_opt():
+    from theano.tensor.nnet import softmax_op
+    from theano.gradient import verify_grad
```
Comment (lamblin) [inline theano/sandbox/cuda/tests/test_dnn.py]: flake8 says: `F401 'verify_grad' imported but unused`

**Comment by abergeron** [comment]
Fixed

**Comment by nouiz** [comment]
Why this is useful? This is already tested in test_nnet.py (it test our softmax and cudnn one).

**Comment by nouiz** [comment]
Maybe we can move the cudnn one to test_dnn.py file

**Comment by abergeron** [comment]
Because they are not on test_dnn.py I didn't find them.  Also, why haven't these tests caught that GpuDnnSoftmaxGrad (x, y, 1, 1) doesn't work?

**Comment by nouiz** [comment]
I think it would be great to move the tests to test_dnn.py. It probably didn't tested that shape. Adding it would be good.

**Comment by abergeron** [comment]
I've moved and consolidated the tests.

**Comment by nouiz** [comment]
tests fail.

File: theano/sandbox/cuda/tests/test_dnn.py [hunk 2]
```diff
@@ -450,43 +452,132 @@ def test_pooling_opt():
...
+    def test_dnn_softmax_grad(self):
+        softmax_op = dnn.GpuDnnSoftmax('bc01', 'accurate', 'channel')
+
+        x_val = numpy.random.normal(0, 1, (3, 4, 2, 5)).astype('float32')
+        x_val2 = numpy.random.normal(0, 1, (3, 4, 1, 1)).astype('float32')
+
+        utt.verify_grad(softmax_op, [x_val])
+
+        # Gradient is broken for (n, c, 1, 1) in v3 rc1
+        if cuda.dnn.version() == (2000, 2000):
```
Comment (nouiz) [inline theano/sandbox/cuda/tests/test_dnn.py]: I would do 

... != (3000, 3000), next version should work, so we don't need to update.

Comment (abergeron) [inline theano/sandbox/cuda/tests/test_dnn.py]: Fixed.

**Comment by abergeron** [comment]
What fails exactly in the tests?  I haven't seen any failures on my end.

**Pull request merged**